pytorch_lightning/trainer/trainer.py

```diff
@@ -185,6 +185,8 @@ def __init__(self,
         # creates a default one if none passed in
         self.early_stop_callback = None
         self.configure_early_stopping(early_stop_callback, logger)
+        if self.enable_early_stop:
+            self.nb_sanity_val_steps = max(1, self.nb_sanity_val_steps)
```

Comment:
maybe `max(1, nb_sanity_val_steps)` since earlier you have
```python
if self.fast_dev_run:
    self.nb_sanity_val_steps = 1
```

Reply:
But exactly by that reason it should be `max(1, self.nb_sanity_val_steps)` :)

We just take the previously defined final `self.nb_sanity_val_steps` and set it to `1` if it is less than `1`.

If we made as you have suggested then `self.nb_sanity_val_steps` will be equal to the user defined value in fast dev run mode, but it should be `1`.

Reply:
Let's not do this.  People need to have the option of turning sanity_val_check off

Reply:
But how then we will check that early stopping will work correctly? (Note that we force this check only if early stopping is turned on.)

Reply:
I understand what you're saying, but restricting EVERYONE to force sanity check will certainly block some esoteric research or production cases, so we can't do this.

But I think this is on the user at this point. If they turned off sanity check then it's on them at that point... but for people who keep it on, then we use what you suggest.

Reply:
Done